### PR TITLE
After long update no exposure, additonal logs & potential fix (EXPSOUREAPP-4120)

### PR DESCRIPTION
**Pull request by BMItr**
#During analysis of 4120 & 4087 I spotted two points which should be adjusted 

This PR addresses (DRAFT):
- Exception while using the work:work-runtime-ktx:2.5.0-beta01 during runtime -> Downgrade to latest stable (2.4.0), which solves the Exception
- Added additonal log to DefaultDiagnosisKeyProvider for deeper analysis
- Avoid multiple downloadtasks by using startedAt as fallback when finishedAt is null to avoid new download in TracingRepository

### CODE REVIEW

File: Corona-Warn-App/build.gradle [hunk 1]
```diff
@@ -318,7 +318,7 @@ dependencies {
     implementation 'androidx.navigation:navigation-ui-ktx:2.2.2'
     implementation 'androidx.legacy:legacy-support-v4:1.0.0'
     implementation 'androidx.preference:preference:1.1.1'
-    implementation 'androidx.work:work-runtime-ktx:2.5.0-beta01'
+    implementation 'androidx.work:work-runtime-ktx:2.4.0'
```
Comment (d4rken) [inline Corona-Warn-App/build.gradle:321]: Are both issues related or why are both changes in the same PR?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/TracingRepository.kt [hunk 2]
```diff
@@ -153,11 +153,10 @@ class TracingRepository @Inject constructor(
         val taskLastFinishedAt = try {
             taskController.tasks.first()
                 .filter { it.taskState.type == DownloadDiagnosisKeysTask::class }
-                .mapNotNull { it.taskState.finishedAt }
-                .sortedDescending()
-                .first()
+                .mapNotNull { if (it.taskState.finishedAt != null) it.taskState.finishedAt else it.taskState.startedAt }
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/TracingRepository.kt:156]: Maybe `it.taskState.finishedAt ?: it.taskState.startedAt`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/TracingRepository.kt [hunk 3]
```diff
@@ -153,11 +153,10 @@ class TracingRepository @Inject constructor(
...
+                .maxOrNull()!!
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/TracingRepository.kt:157]: Doesn't this now throw a `KotlinNullpointerException` which will no longer be caught by `catch(e: NoSuchElementException)`?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/TracingRepository.kt (hunk 2 above)
Comment (BMItter) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/TracingRepository.kt]: good point. adjusted.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/TracingRepository.kt (hunk 3 above)
Comment (BMItter) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/TracingRepository.kt:156]: adjusted to catch Nullpointer

File: Corona-Warn-App/build.gradle (hunk 1 above)
Comment (BMItter) [reply Corona-Warn-App/build.gradle:321]: tickets were declared to related inside jira.
I can separte them if needed.

**Review by ralfgehrer — APPROVED** [review]

**Review by d4rken — APPROVED** [review]

**Pull request merged**